pkg/workspace/controllers/workspace_controller.go

```diff
@@ -525,17 +525,8 @@ func (c *WorkspaceReconciler) applyInference(ctx context.Context, wObj *kaitov1b
 				return
 			}
 
-			// Assign the correct type to existingObj based on the type of workloadObj.
-			var existingObj client.Object
-			var desiredPodSpec *corev1.PodSpec
-			switch workloadObj := workloadObj.(type) {
-			case *appsv1.StatefulSet:
-				existingObj = &appsv1.StatefulSet{}
-				desiredPodSpec = &workloadObj.Spec.Template.Spec
-			case *appsv1.Deployment:
-				existingObj = &appsv1.Deployment{}
-				desiredPodSpec = &workloadObj.Spec.Template.Spec
-			}
+			existingObj := &appsv1.StatefulSet{}
```

Comment:
so this PR would convert all deployments to statefulsets even for small models? and some azure vm GPU sku does not have local nvme disk, what would be the volume created on those skus?